### ISSUE TITLE
[Feature] topk selection operation

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#35]
## Summary

Add `TopKOp` to select the k largest (or smallest) values and their indices along a specified axis.

## Motivation

Top-K selection is essential for classification post-processing (e.g., "top 5 predictions"), beam search in NLP, and NMS-based detection pipelines. It complements argmax for cases where multiple top predictions are needed.

## Reference Implementations

- **PyTorch**: `torch.topk(input, k, dim=-1, largest=True, sorted=True)` — [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.topk.html)
- **ONNX**: `TopK` operator (opset 1+) — [ONNX spec](https://onnx.ai/onnx/operators/onnx__TopK.html)

## Verification

- ✅ `torch.topk` exists — VERIFIED (returns namedtuple of values and indices)
- ✅ ONNX `TopK` operator exists
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for topk, top_k, topK)

## Proposed API

```dart
// Top 5 along last dimension
final (values, indices) = tensor.topk(5);

// Top 3 along axis 1, sorted
final (values, indices) = tensor.topk(3, axis: 1, largest: true, sorted: true);

// As TransformOp
final op = TopKOp(k: 5, axis: -1, largest: true, sorted: true);
```

## Implementation Notes

- Return a tuple/record of `(TensorBuffer values, TensorBuffer indices)`
- Support `largest` flag (true = largest, false = smallest)
- Support `sorted` flag
- Indices should be `DType.int64`
- Use partial sort (quickselect) for O(n + k log k) efficiency
- Add `OperationCapabilities`:
  - `onnxOpType: 'TopK'`
  - `pytorchEquivalent: 'torch.topk'`

## ROI Assessment

**High** — Critical for multi-label classification output decoding and detection pipelines. Works well with existing softmax activation and the argmax feature (separate issue).
